webapp/src/js/app.js

```diff
@@ -157,6 +157,11 @@ const createReduxLoggerConfig = Selectors => ({
     $ngReduxProvider.createStoreWith(RootReducer, middlewares);
   });
 
+  // 16 million characters is gauranteed to be less than the API JSON
+  // parser limit of 32MB. It's possible to work out the exact size of
+  // the body based on the number of two byte characters but this is
+  // expensive.
+  const BODY_LENGTH_LIMIT = 16000000; // 16 million
```

Comment:
REVIEWER: Do you think it's worth calculating the exact size of the request? And what about these numbers - should it be 32M chars and 64MB in API?